changelogs/2.10.0.md

```diff
@@ -0,0 +1,933 @@
+  and `select()`, in case those were called without specifying a timeout
+  (gh-6242).
+
+* Improved `net.box` performance by up to 70% by rewriting hot code paths in C
+  (gh-6241).
+* Introduced compact tuples that allow saving 4 bytes per tuple in case of
+  small userdata (gh-5385).
+* Streams and interactive transactions over streams are implemented
+  in iproto. A stream is associated with its ID, which is unique within
+  one connection. All requests with the same not zero stream ID belong to
+  the same stream. All requests in the stream are processed synchronously.
+  The execution of the next request will not start until the previous one is
+  completed. If a request has zero stream ID, it does not belong to a stream
+  and is processed in the old way.
+
+  In `net.box`, a stream is an object above connection that has the same
+  methods but allows to execute requests sequentially. ID is generated
+  automatically on the client side. If users write their own connector and
+  want to use streams, they must transmit stream_id over iproto protocol.
+  The main purpose of streams is transactions via iproto.
+  Each stream can start its own transaction, allowing multiplexing several
+  transactions over one connection. There are multiple ways to begin,
+  commit and rollback a transaction: using appropriate stream methods, using
+  `call` or `eval` methods or using `execute` method with SQL transaction
+  syntax. Users can mix these methods, for example, start a transaction using
+  `stream:begin()`, and commit a transaction using `stream:call('box.commit')`
+  or `stream:execute('COMMIT')`.
+
+  If any request fails during the transaction, it will not affect the other
+  requests in the transaction. If a disconnect occurs when there is some active
+  transaction in the stream, this transaction will be rolled back if it
+  does not have time to commit before this moment.
+* Added new `memtx_allocator` option to `box.cfg{}` that allows to select the
+  appropriate allocator for memtx tuples if necessary. Possible values are
+  `system` for malloc allocator and `small` for default small allocator.
+
+  Implemented system allocator, based on malloc: slab allocator that is used for
+  tuples allocation has a certain disadvantage---it tends to unresolvable
+  fragmentation on certain workloads (size migration). In this case, a user
+  should be able to choose another allocator. System allocator based on malloc
+  function, but restricted by the same quota as slab allocator. System
+  allocator does not alloc all memory at start. Instead, it allocates memory as
+  needed, checking that quota is not exceeded (gh-5419).
+* Added `box.stat.net.thread()` for reporting per thread net statistics
+  (gh-6293).
+* Added the new `STREAMS` metric to `box.stat.net` that contains statistics
+  for iproto streams. `STREAMS` contains same counters as `CONNECTIONS`
+  metric in `box.stat.net`: current, rps, and total (gh-6293).
```

Comment:
shouldn't it be RPS? https://en.wikipedia.org/wiki/Requests_per_second#requests_per_second